hack/generate/gen-test-framework.sh

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+
+#############################
+# Make sure that test/test-framework is updated according to the current source code
+#############################
+
+set -ex
+
+source hack/lib/test_lib.sh
+
+# Define vars
+DOCKERFILE="build/Dockerfile"
+GOMOD="go.mod"
+
+# Go inside of the mock data test project
+cd test/test-framework
+
+# Create the required files in the mock test/test-framework just to allow run from it the oprator-sdk commands
+echo > $DOCKERFILE
+echo > $GOMOD
+
+# Run gen commands
+GO111MODULE=on operator-sdk generate k8s
+GO111MODULE=on operator-sdk generate openapi
+
+# Remove files that are required and created just to exec the commands.
+trap_add 'rm -rf $DOCKERFILE' EXIT
+trap_add 'rm -rf $GOMOD' EXIT
+trap_add 'rm -rf go.sum' EXIT
```

Comment:
Move these lines above the `operator-sdk generate` calls in 23 and 24. `trap_add` is kind of like `defer` in Go. It's basically saying, "when the script exits, execute this command". We need to set the trap up _before_ we run the commands that may fail to guarantee that these files are cleaned up.